SOURCE/FD_Jacobian.c

```diff
@@ -67,38 +67,27 @@ void PhaseRheologyLoop( int centroid, double sign, double denom, double Exx, dou
         }
         
         if ( model->eta_avg == 0) { // ARITHMETIC AVERAGE
-            if ( cond == 1 ) detadE[c0]      += sign*vol[p][c0] * etaVE/(denom);
+            if ( cond == 1 ) detadE[c0]      += sign*vol[p][c0] * etaVE/(denom); // TODO: reformat etaVE to eta_eff
         }
         
         if ( model->eta_avg == 1) { // HARMONIC AVERAGE
-            if ( cond == 1 ) detadE[c0]      += sign*vol[p][c0] * etaVE/(denom) / pow(etaVE,2.0);
+            if ( cond == 1 ) detadE[c0]      += sign*vol[p][c0] * etaVE/(denom) / pow(eta_phase[p][c0],2.0);
         }
         
         if ( model->eta_avg == 2) { // GEOMETRIC AVERAGE
-            if ( cond == 1 ) detadE[c0]      += sign*vol[p][c0] * etaVE/(denom) / etaVE;
+            if ( cond == 1 ) detadE[c0]      += sign*vol[p][c0] * etaVE/(denom) / eta_phase[p][c0];
         }
         // OTHERs
         if (centroid > 0) { // if centroid, compute all derivatives of divp
```

Comment:
When we use integers for specifying modes or types, we can name those variables clearer.
like centroid_type

Reply:
changed to `is_centroid`